app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -1965,6 +2004,36 @@ class MessagingManager @Inject constructor(
         }
     }
 
+    private suspend fun setAppLock(data: Map<String, String>) {
+        val app_lock_enable_value = data[APP_LOCK_ENABLED]?.toLowerCase()?.toBooleanStrictOrNull()
+        val app_lock_timeout_value = data[APP_LOCK_TIMEOUT]?.toIntOrNull()
+        val home_bypass_value = data[APP_LOCK_ENABLED]?.toLowerCase()?.toBooleanStrictOrNull()
+
+        var updated = false
+        if (app_lock_enable_value != null) {
+            val canAuth = (BiometricManager.from(context).canAuthenticate() == BiometricManager.BIOMETRIC_SUCCESS)
+            if (app_lock_enable_value && !canAuth) {
+                Log.w(TAG, "Not enabling app lock. BiometricManager cannot Authenticate!")
+            } else {
+                authenticationUseCase.setLockEnabled(app_lock_enable_value)
+                updated = true
+            }
+        }
+        if (app_lock_timeout_value != null && app_lock_timeout_value >= 0) {
+            integrationUseCase.sessionTimeOut(app_lock_timeout_value)
+            updated = true
+        }
+        if (home_bypass_value != null) {
+            authenticationUseCase.setLockHomeBypassEnabled(home_bypass_value)
+            updated = true
+        }
+
+        if (!updated) {
+            Log.w(TAG, "No app lock parameters have been updated")
+            sendNotification(data)
+        }
```

Comment:
Is this still necessary now that we do the check earlier?

Reply:
It's still possible that no biometric authentication can be performed. But looking at it again it could be refactored into something cleaner since earlier checks are now performed.

I'll see if I get a chance to update this some time tomorrow.